code/modules/mob/living/carbon/human/species_types/synths.dm

```diff
@@ -1,37 +1,86 @@
-/datum/species/synth
-	name = "Synth" //inherited from the real species, for health scanners and things
+#define isipcperson(A) (is_species(A, /datum/species/synth))
+GLOBAL_LIST_EMPTY(ipc_screens_list)
+GLOBAL_LIST_EMPTY(ipc_antennas_list)
+
+/datum/species/synth // im fucking lazy mk2 and cant get sprites to normally work
+	name = "IPC" //inherited from the real species, for health scanners and things
 	id = "synth"
 	say_mod = "beep boops" //inherited from a user's real species
 	sexes = 0
-	species_traits = list(NOTRANSSTING) //all of these + whatever we inherit from the real species
-	inherent_traits = list(TRAIT_VIRUSIMMUNE,TRAIT_NODISMEMBER,TRAIT_NOLIMBDISABLE,TRAIT_NOHUNGER,TRAIT_NOBREATH)
+	species_traits = list(NOTRANSSTING,NOBLOOD,TRAIT_EASYDISMEMBER,NOFLASH) //all of these + whatever we inherit from the real species
+	inherent_traits = list(TRAIT_RESISTCOLD,TRAIT_VIRUSIMMUNE,TRAIT_NOLIMBDISABLE,TRAIT_NOHUNGER,TRAIT_NOBREATH,TRAIT_RADIMMUNE,TRAIT_LIMBATTACHMENT)
 	inherent_biotypes = list(MOB_ROBOTIC, MOB_HUMANOID)
 	meat = null
+	exotic_blood = "oil"
 	damage_overlay_type = "synth"
 	limbs_id = "synth"
-	var/disguise_fail_health = 75 //When their health gets to this level their synthflesh partially falls off
-	var/datum/species/fake_species //a species to do most of our work for us, unless we're damaged
+	mutant_bodyparts = list("ipc_screen", "ipc_antenna")
+	default_features = list("mcolor" = "0F0","ipc_screen" = "BSOD", "ipc_antenna" = "None")
+	burnmod = 1.75
+	heatmod = 1.6
+	brutemod = 1.2
 	var/list/initial_species_traits //for getting these values back for assume_disguise()
 	var/list/initial_inherent_traits
 	changesource_flags = MIRROR_BADMIN | WABBAJACK
-
-/datum/species/synth/New()
-	initial_species_traits = species_traits.Copy()
-	initial_inherent_traits = inherent_traits.Copy()
-	..()
+	var/datum/action/innate/monitor_change/screen
+
+/datum/species/synth/spec_emp_act(mob/living/carbon/human/H, severity)
+	. = ..()
+	switch(severity)
+		if(1)
+			H.Stun(160)
+			H.adjustBruteLoss(50)
+		if(2)
+			H.Stun(60)
+			H.adjustBruteLoss(35)
+
+/datum/species/synth/check_roundstart_eligible()
+	return TRUE
+
+/datum/species/synth/military/check_roundstart_eligible()
+	return FALSE
+
+/datum/species/synth/spec_attacked_by(obj/item/I, mob/living/user, obj/item/bodypart/affecting, intent, mob/living/carbon/human/H)

```

Comment:
This proc is a mess.

Reply:
which proc, the first one or second?

Reply:
The entirety of `/datum/species/synth/spec_attacked_by(`

Reply:
spaghetticode galore, robot welding/wiring standard didnt work, and this is a way to assure they can self repair, and be repaired is that

Reply:
No like there's actual code issues you need to address. Throughout the proc.

Reply:
could you please point them out because it runs fine on my PC and runs fine on toxics, and i don't get any compile errors

then again i don't code so i'm not the best at seeing them

blease ike

Reply:
Yeah it's obvious you don't code. Here is where I rewrote it to not be terrible: https://pastebin.com/WxDmL7Zh